src/codeFlowLogin.ts

```diff
@@ -14,12 +14,13 @@ import * as net from 'net';
 import * as vscode from 'vscode';
 import { Environment } from '@azure/ms-rest-azure-env';
 import { TokenResponse, AuthenticationContext } from 'adal-node';
+import { AzureAccountEnvironment } from './azure-account.api';
 
 export const redirectUrlAAD = 'https://vscode-redirect.azurewebsites.net/';
 const portADFS = 19472;
 const redirectUrlADFS = `http://127.0.0.1:${portADFS}/callback`;
 
-export function isADFS(environment: Environment) {
+export function isADFS(environment: AzureAccountEnvironment) {
```

Comment:
Since the environment itself now has this flag, can you remove this function and use environment.isAzureStack or environment.azureStackApiProfile in its place?

Reply:
Hey Rachel, I am not sure if we can get rid of this. Since Azure Stack can be either AAD or ADFS. I believe we still need ADFS check for us. In https://github.com/microsoft/vscode-azure-account/blob/4f1a49ec5fe4f7dec21b93bc736cf2f3af61cb3e/src/codeFlowLogin.ts#L189. I believe we will miss this if we don't do ADFS check.

Reply:
Yes, AzureStack environment can be configured either with AAD (connected) or ADFS (disconnected)